### PR TITLE
Utilize core-based synchronisation

### PR DESCRIPTION
**Pull request by chezzdev**
Checks:

* [ ] Update changelog
* [x] Rebase to `dev` branch
* [x] Assign reviewers

Linked issues:
Another iOS part of #1739

### CODE REVIEW

File: MapboxVision/Services/Recording/VideoTrimmer.swift [hunk 1]
```diff
@@ -41,23 +36,28 @@ final class VideoTrimmer {
             return
```
Comment (dersim-davaod) [inline MapboxVision/Services/Recording/VideoTrimmer.swift]: `completion` is not called for else-branch of guard statement here and line 36 above.
IMHO we should call it.

File: MapboxVision/Services/Recording/VideoTrimmer.swift [hunk 2]
```diff
@@ -41,23 +36,28 @@ final class VideoTrimmer {
...
             try videoTrack.insertTimeRange(timeRangeForCurrentSlice, of: videoAssetTrack, at: CMTime())
         } catch {
-            completion?(error)
+            completion(error)
         }
 
-        guard let exportSession = AVAssetExportSession(asset: composition, presetName: preferredPreset) else { return }
+        guard
+            let exportSession = AVAssetExportSession(asset: composition, presetName: AVAssetExportPresetPassthrough)
+        else { return }
```
Comment (dersim-davaod) [inline MapboxVision/Services/Recording/VideoTrimmer.swift]: same concern about `completion` which is not called on else-branch.

File: MapboxVision/Services/Recording/VideoRecorder.swift [hunk 3]
```diff
@@ -50,7 +56,7 @@ final class VideoRecorder {
     }
 
     func stopRecording(completion: (() -> Void)?) {
-        writerQueue.async { [weak self] in
+        writerQueue.sync { [weak self] in
```
Comment (dersim-davaod) [inline MapboxVision/Services/Recording/VideoRecorder.swift]: Do we need to weakify self?

Comment (chezzdev) [reply MapboxVision/Services/Recording/VideoRecorder.swift]: Now that it's `sync`, no. Will update.

File: MapboxVision/Services/Recording/VideoTrimmer.swift (hunk 1 above)
Comment (chezzdev) [reply MapboxVision/Services/Recording/VideoTrimmer.swift]: Good catch. Will update.

File: MapboxVisionTests/DeviceCheckerTests.swift [hunk 4]
```diff
@@ -89,6 +89,18 @@ class DeviceCheckerTests: XCTestCase {
         // Given iPhone XR
         // When // Then
         XCTAssertTrue(UIDeviceIphoneXRStub().isHighPerformance)
+
```
[the remark this answers is not in the document]
Comment (chezzdev) [reply MapboxVisionTests/DeviceCheckerTests.swift]: I was cleaning the tests, so for them to pass, I've added these cases.

**Pull request merged**